### PR TITLE
Fix reply builder access issue

### PR DESCRIPTION
**Pull request by dranikpg**
This test should trigger the runtime error. Fixes https://github.com/dragonflydb/dragonfly/issues/782#issuecomment-1581227708

from #782 issue

### CODE REVIEW

**Comment by dranikpg** [comment]
@romange 

This cases a runtime error, but doesn't crash the instance. How do you suggest to assert check pointer validity? (given that we'll implement a fix for it)

File: tests/dragonfly/connection_test.py [hunk 1]
```diff
@@ -346,3 +347,51 @@ async def test_subscribe_in_pipeline(async_client: aioredis.Redis):
...
     assert res == ['one', ['subscribe', 'ch1', 1], 'two', ['subscribe', 'ch2', 2], 'three']
+
+"""
+This test makes sure that Dragonfly can receive blocks of pipelined commands even
+while a script is still executing. This is a dangerous scenario because both the dispatch fiber
+and the connection fiber are actively using the context. What is more, the script execution injects
+its own custom reply builder, which can't be used anywhere else, besides the lua script itself.
+"""
+
+async def test_parser_while_script_running(async_client, df_server):
```
Comment (romange) [inline tests/dragonfly/connection_test.py:358]: please add types to variables.

File: tests/dragonfly/connection_test.py [hunk 2]
```diff
@@ -346,3 +347,51 @@ async def test_subscribe_in_pipeline(async_client: aioredis.Redis):
...
+    P2 = """
+    MGET m1 m2 m3
+    MGET m4 m5 m6
+    MGET m7 m8 m9\n
+    """
+    P3 = """
+    PING
+    """ * 50 + "ECHO DONE\n"
+
+    c = aioredis.Redis(port=df_server.port)
```
Comment (romange) [inline tests/dragonfly/connection_test.py:379]: why using `c` if you have an `async_client`?

File: tests/dragonfly/connection_test.py [hunk 3]
```diff
@@ -346,3 +347,51 @@ async def test_subscribe_in_pipeline(async_client: aioredis.Redis):
...
+    sha = await c.script_load(S)
+
+    reader, writer = await asyncio.open_connection('localhost', df_server.port)
+
+    writer.write(P1.format(sha=sha).encode())
+    await writer.drain()
+
+    await asyncio.sleep(0.01)
+
+    writer.write(P2.encode())
```
Comment (romange) [inline tests/dragonfly/connection_test.py:389]: why you can not just sending the commands via pipeline mode?

File: tests/dragonfly/connection_test.py [hunk 4]
```diff
@@ -346,3 +347,51 @@ async def test_subscribe_in_pipeline(async_client: aioredis.Redis):
...
+    writer.write(P3.encode())
+    await writer.drain()
+
+    out = await reader.readuntil(b"DONE")
+    print(out)
+
+    writer.close()
```
Comment (romange) [inline tests/dragonfly/connection_test.py:396]: I do not think we need to check or assert anything. This test covers a specific use-case that crashes DF in debug mode. It's interesting enough already

File: src/facade/dragonfly_connection.cc [hunk 5]
```diff
@@ -563,6 +563,9 @@ void Connection::DispatchCommand(uint32_t consumed, mi_heap_t* heap) {
 }
 
 Connection::ParserStatus Connection::ParseRedis() {
+  VLOG(0) << "ParseRedis " << int64_t(cc_.get()) << " ptr " << int64_t(cc_->reply_builder())
```
Comment (romange) [inline src/facade/dragonfly_connection.cc:566]: revert this line

**Comment by dranikpg** [comment]
Omg, I didn't ask for a review 😧 But now you can take a look at it

I've failed to crash the pytest on CI. It fails on my local computer and on different AWS instances, but not there.

**Comment by dranikpg** [comment]
With the current test range it fails almost certainly for all thread counts (<=8) on my machine

**Review by romange — APPROVED** [review]

**Pull request merged**